src/02-common-principles.md

```diff
@@ -6,56 +6,72 @@ The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
-    field maps), anat (structural imaging such as T1, T2, etc.), meg
-    (magnetoencephalography), eeg (electroencephalography), ieeg (intracranial
-    electroencephalography), beh (behavioral).
+1.  **Data type** - a functional group of different types of data. In BIDS we
+    define eight data types: `func` (task based and resting state functional MRI),
+    `dwi` (diffusion weighted imaging), `fmap` (field inhomogeneity mapping data
+    such as field maps), `anat` (structural imaging such as T1, T2, etc.), `meg`
+    (magnetoencephalography), `eeg` (electroencephalography), `ieeg` (intracranial
+    electroencephalography), `beh` (behavioral).
 
-1.  Task - a set of structured activities performed by the participant. Tasks
-    are usually accompanied by stimuli and responses, and can greatly vary in
-    complexity. For the purpose of this specification we consider the so-called
+1.  **Task** - a set of structured activities performed by the participant.
+    Tasks are usually accompanied by stimuli and responses, and can greatly vary
+    in complexity. For the purpose of this specification we consider the so-called
     "resting state" a task. In the context of brain scanning, a task is always
     tied to one data acquisition. Therefore, even if during one acquisition the
     subject performed multiple conceptually different behaviors (with different
     sets of instructions) they will be considered one (combined) task.
 
-1.  Event - a stimulus or subject response recorded during a task. Each event
-    has an onset time and duration. Note that not all tasks will have recorded
-    events (e.g., resting state).
+1.  **Event** - a stimulus or subject response recorded during a task. Each
+    event has an onset time and duration. Note that not all tasks will have
+    recorded events (e.g., “resting state”).
 
-1.  Run - an uninterrupted repetition of data acquisition that has the same
+1.  **Run** - an uninterrupted repetition of data acquisition that has the same
     acquisition parameters and task (however events can change from run to run
     due to different subject response or randomized nature of the stimuli). Run
     is a synonym of a data acquisition.
 
+1.  **`<index>`** - a numeric value, possibly prefixed with arbitrary number of
+    0s for consistent indentation, e.g., it is `01` in `run-01` following
+    `run-<index>` specification.
+
+1.  **`<label>`** - an alphanumeric value, possibly prefixed with arbitrary
+    number of 0s for consistent indentation, e.g., it is `rest` in `task-rest`
+    following `task-<label>` specification.
+
+1.  **`suffix`** - a portion of the file name with `key-value_` pairs (thus after
+    the final `_`), right before the **File extension**.
+
+1.  **File extension** - a portion of the the file name after the left-most
+    period (`.`) preceded by any other alphanumeric (so `.gitignore` does not have a
+    suffix)
```

Comment:
```suggestion
1.  **File extension** - a portion of the the file name after the left-most
    period (`.`) preceded by any other alphanumeric. For example, `.gitignore` does
    not have a file extension, but the file extension of `test.nii.gz` is `.nii.gz`.
    Note that the left-most period is included in the file extension.
```

> Python seems to also retain period in splitext 
> makes it easier to parse when period is there

I agree with these two arguments. But if pybids does it differently, we may go with how pybids does it. I am not sure ...

Reply:
In MATLAB the period is part of the file extension as well. 
```
>> [p, f, x] = fileparts('/a/b/c.d')
p =
    '/a/b'
f =
    'c'
x =
    '.d'
```

Reply:
I am torn on this one, but I think that so far it seems that PyBIDS is unfortunately the only argument against including `.`... I think it would be relatively easy to support "official" way with minor compatibility shim on pybids side. Additional minor factor against adding it is added cumbersomness of defining regex based matching (now it would need to include into the group the `.` but also then have another group for alternative extensions e.g. `(?P<extension>.(ext1|ext2))` in Python).

Overall I think we better include '.' to stay inline with current wording etc of BIDS official specification

Reply:
crap, committed the other suggestion with the wrong commit message.  I will accept it now here and then force push rewrite with corrected messages.